src/Symfony/Component/Serializer/Encoder/XmlEncoder.php

```diff
@@ -376,7 +379,9 @@ private function buildXml(\DOMNode $parentNode, $data, string $xmlRootNodeName =
                     if (!is_scalar($data)) {
                         $data = $this->serializer->normalize($data, $this->format, $this->context);
                     }
-                    $parentNode->setAttribute($attributeName, $data);
+                    if (null !== $data || !$removeEmptyAttributes) {
```

Comment:
But if `$data` is `''` (an empty string), the empty attribute will still be added. It's why I was suggesting to reuse the existing `skip_null_values` context option instead `remove_empty_attributes` which is misleading (because empty attributes aren't removed).